### PR TITLE
Add load table settings

### LINKED ISSUE #26: Passing options to parquet_handler, data_cleaner, etc. from the top API
Currently, only specific classes used internally take options.

Perhaps we could pass each classes options/initial instance directly to the top level functions?

### PR DESCRIPTION
**Pull request by m-amin-alavian**
This PR implements a LoadTable settings model to configure data loading behavior.

The LoadTable model in metadata.py encapsulates options like:

Data type (original, cleaned, processed)
Missing file handling (error, download, create)
Caching flags
The settings.yaml file has been updated to include a load_table section for defaults.

Functions like load_table() now accept a LoadTable object to specify options.

Closes #26

### CODE REVIEW

File: hbsir/__init__.py [hunk 1]
```diff
@@ -1,27 +1,65 @@
 """init file"""
 
-from typing import Iterable
+from typing import Iterable, Literal
 
 import pandas as pd
 
-from . import data_engine
+from . import data_engine, metadata, data_cleaner, utils
```
[the remark this answers is not in the document]
Comment (TerseTears) [reply hbsir/__init__.py:7]: Perhaps it would be easier to merge this branch first and update the other branch.

File: hbsir/metadata.py [hunk 2]
```diff
@@ -96,6 +107,23 @@ class Metadatas:
...
     other = open_yaml("metadata/other.yaml")
 
+    def reload_schema(self):
+        self.schema = open_yaml("metadata/schema.yaml")
+
+
+load_table_defaults = settings["functions_defaults"]["load_table"]
+
+
+class LoadTable(BaseModel):
```
Comment (TerseTears) [inline hbsir/metadata.py:117]: Shouldn't this be dataclass?

File: hbsir/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-amin-alavian) [reply hbsir/__init__.py:7]: OK

File: hbsir/metadata.py (hunk 2 above)
Comment (m-amin-alavian) [reply hbsir/metadata.py:117]: It seems that it is possible to use dataclass in Pydantic, as mentioned in their documentation: https://docs.pydantic.dev/latest/usage/dataclasses/.
But it is probably better to consider this later.

**Pull request merged**